addons/mrp/migrations/15.0.2.1/pre-migration.py

```diff
@@ -0,0 +1,18 @@
+def migrate(cr, version):
+    cr.execute(
+        """
+        WITH subquery AS (
+            SELECT mp.id,
+                   MIN(wo.date_start) AS date_start
+              FROM mrp_production mp
+              JOIN mrp_workorder wo
+                ON wo.production_id = mp.id
+             WHERE mp.state != 'draft'
+               AND mp.date_start IS NULL
+             GROUP BY mp.id
+        ) UPDATE mrp_production mp
+             SET date_start = subquery.date_start
+            FROM subquery
+           WHERE mp.id = subquery.id
```

Comment:
That is CTE, not Subquery. It's better to name it by business logics
```suggestion
        WITH production_info AS (
            SELECT mp.id,
                   MIN(wo.date_start) AS date_start
              FROM mrp_production mp
              JOIN mrp_workorder wo
                ON wo.production_id = mp.id
             WHERE mp.state != 'draft'
               AND mp.date_start IS NULL
             GROUP BY mp.id
        )
        UPDATE mrp_production mp
            SET date_start = info.date_start
          FROM production_info info
          WHERE mp.id = info.id
```

Reply:
@davidtranhp yes sir